lib/check-port.js

```diff
@@ -0,0 +1,31 @@
+const tcpPortUsed = require('tcp-port-used');
+const chalk = require('chalk');
+
+function checkPort(port, content, numOfAttempts, MAX_ATTEMPTS) {
```

Comment:
why do we need to put MAX_ATTEMPTS as function argument everytime we call the function ?
1. extract out MAX_ATTEMPTS since it's constant
```js
constant MAX_ATTEMPTS = 10
```

```js
function checkPort(port, content, numOfAttempts) {
```

Reply:
Because this function is repeatedly called by both `start-server` and `server` so the value of `MAX_ATTEMPTS` is increased by both the function so we should increment it locally by each so that `start-server` port and `live-reload` port is incremented 10 times each.

Reply:
```js
function checkPort(port, content, numOfAttempts, MAX_ATTEMPTS) {
  tcpPortUsed
    .check(port, 'localhost')
    .then(function(inUse) {
      if (inUse && numOfAttempts >= MAX_ATTEMPTS) {
        console.log(
          'Reached max attempts, exiting. Please open up some ports or ' +
            'increase the number of attempts and try again.'
        );
        process.exit(1);
      } else if (inUse) {
        console.error(chalk.red('Port ' + port + ' is in use'));
        // Try again but with port + 1
        port += 1;
        numOfAttempts += 1;
        checkPort(port, content);
      } else {
        content(port);
      }
    })
    .catch(function(ex) {
      setTimeout(function() {
        throw ex;
      }, 0);
    });
}
```

You are only increasing `numOfAttempts` though

Reply:
oops!! My Bad 😆 , Was confused :)